pinot-common/src/main/java/org/apache/pinot/common/request/transform/TransformExpressionTree.java

```diff
@@ -75,6 +80,36 @@ public static String getStandardExpression(AstNode astNode) {
     }
   }
 
+  public static Expression getExpression(AstNode astNode) {
+    Expression expression = new Expression();
+    if (astNode instanceof IdentifierAstNode) {
+      // Column name
+      expression.setType(org.apache.pinot.common.request.ExpressionType.IDENTIFIER);
+      expression.setIdentifier(new Identifier(((IdentifierAstNode) astNode).getName()));
+    } else if (astNode instanceof FunctionCallAstNode) {
+      // UDF expression
+      Function func = new Function(((FunctionCallAstNode) astNode).getName().toLowerCase());
+      if (astNode.getChildren()!=null) {
+        for (AstNode child : astNode.getChildren()) {
+          func.addToOperands(getExpression(child));
+        }
+      }
+      expression.setType(org.apache.pinot.common.request.ExpressionType.FUNCTION);
+      expression.setFunctionCall(func);
+    } else if (astNode instanceof StringLiteralAstNode) {
+      // Treat string as column name
+      // NOTE: this is for backward-compatibility
+      expression.setType(org.apache.pinot.common.request.ExpressionType.IDENTIFIER);
```

Comment:
hmm, let's keep this part clean. We can handle backward compatibility in the converter.